daemon/algod/api/server/v2/handlers.go

```diff
@@ -838,6 +838,40 @@ func (v2 *Handlers) GetBlockHash(ctx echo.Context, round uint64) error {
 	return ctx.JSON(http.StatusOK, response)
 }
 
+// GetBlockHeader gets the block header for the given round.
+// (GET /v2/blocks/{round}/header)
+func (v2 *Handlers) GetBlockHeader(ctx echo.Context, round uint64, params model.GetBlockHeaderParams) error {
+	handle, contentType, err := getCodecHandle((*string)(params.Format))
+	if err != nil {
+		return badRequest(ctx, err, errFailedParsingFormatOption, v2.Log)
+	}
+
+	ledger := v2.Node.LedgerForAPI()
+	block, err := ledger.BlockHdr(basics.Round(round))
+	if err != nil {
+		switch err.(type) {
+		case ledgercore.ErrNoEntry:
+			return notFound(ctx, err, errFailedLookingUpLedger, v2.Log)
+		default:
+			return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
+		}
+	}
+
+	// Encoding wasn't working well without embedding "real" objects.
+	response := struct {
```

Comment:
Should this be a defined type somewhere, or do we think single use-case/good enough?

Reply:
Sorry for the delay in responding. I'm happy to define a `BlockHeaderResponse` struct above the method if preferred. I saw `getPendingTransactions` and `GetTransactionGroupLedgerStateDeltasForRound` using this anonymous struct approach, and since it's only used here thought it should be OK.